lib/cocoapods/installer/analyzer.rb

```diff
@@ -75,11 +75,15 @@ def analyze(allow_fetches = true)
         store_existing_checkout_options
         fetch_external_sources if allow_fetches
 
-        @locked_dependencies    = generate_version_locking_dependencies
-        @result.specs_by_target = validate_platforms(resolve_dependencies)
-        @result.specifications  = generate_specifications
-        @result.targets         = generate_targets
+        @locked_dependencies = generate_version_locking_dependencies
+        resolver_specs_by_target = resolve_dependencies
+        validate_platforms(resolver_specs_by_target)
+        @result.specifications  = generate_specifications(resolver_specs_by_target)
+        @result.targets         = generate_targets(resolver_specs_by_target)
         @result.sandbox_state   = generate_sandbox_state
+        @result.specs_by_target = resolver_specs_by_target.each_with_object({}) do |rspecs_by_target, hash|
```

Comment:
The result outside of the `analyzer` remains the same as it used to. A hash of specs by target.